newIDE/app/src/MainFrame/index.js

```diff
@@ -351,6 +364,11 @@ const MainFrame = (props: Props) => {
     setFileMetadataOpeningMessage,
   ] = React.useState<?MessageDescriptor>(null);
 
+  // This reference is useful to get the current opened project.
+  // It might be different from the current project an effect or a callback
+  // manipulates when a promise resolves for instance.
+  const currentProjectRef = React.useRef<?gdProject>(null);
```

Comment:
```suggestion
  /* This reference is useful to get the current opened project, even in the callback of a hook/promise - without risking to read "stale" data .
   * This can be different from the `currentProject` (coming from the state) that an effect or a callback
   * manipulates when a promise resolves for instance.
   * See `isCurrentProjectStale`.
   */
  const currentProjectRef = React.useRef<?gdProject>(null);
```